src/handlers/gestures/GestureDetector/useViewRefHandler.ts

```diff
@@ -0,0 +1,55 @@
+import { findNodeHandle } from '../../gestureHandlerCommon';
+import { validateDetectorChildren } from './utils';
+
+import { isFabric, tagMessage } from '../../../utils';
+import { getShadowNodeFromRef } from '../../../getShadowNodeFromRef';
+
+import { GestureDetectorState } from './types';
+import React, { useCallback } from 'react';
+
+declare const global: {
+  isFormsStackingContext: (node: unknown) => boolean | null; // JSI function
+};
+
+// Ref handler for the Wrap component attached under the GestureDetector.
+// It's responsible for setting the viewRef on the state and triggering the reattaching of handlers
+// if the view has changed.
+export function useViewRefHandler(
+  state: GestureDetectorState,
+  updateAttachedGestures: (skipConfigUpdate?: boolean) => void
+) {
+  const refHandler = useCallback(
+    (ref: React.Component | null) => {
+      if (ref === null) {
+        return;
+      }
+
+      validateDetectorChildren(ref);
```

Comment:
Is it necessary? As far as I can see it is extracted from [this](https://github.com/software-mansion/react-native-gesture-handler/blob/50d2dce0ef1e70c8baf956815bb156c7f08613f3/src/handlers/gestures/GestureDetector.tsx#L819), where there is no validation performed.

Reply:
You're right, it's not. The validation is already done in `updateAttachedHandlers`.